docs/cluster_and_device_type_dev/how_to_add_new_dts_and_clusters.md

```diff
@@ -0,0 +1,248 @@
+        2. Check that you have added your cluster code into
+           [mode-base-cluster.xml](https://github.com/project-chip/connectedhomeip/tree/master/src/app/zap-templates/zcl/data-model/chip/mode-base-cluster.xml)
+            1. `<struct name="ModeTagStruct"> <cluster code="0xXXXX">` - replace
+               `XXXX` with your cluster ID
+            2. `<struct name="ModeOptionStruct"> <cluster code="0xXXXX">` -
+               replace `XXXX` with your cluster ID
+            3. `<bitmap name="Feature" type="bitmap32"> <cluster code="0xXXXX">` -
+               replace `XXXX` with your cluster ID
+        3. Add your new Mode definitions in `xyz-mode.h`. In this header you
+           define the modes / labels and tags. See
+           [dishwasher-mode.h](https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/all-clusters-common/include/dishwasher-mode.h)
+           as an example.
+        4. Add your new stub to instantiate the mode. See
+           [dishwasher-mode.cpp](https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/all-clusters-common/src/dishwasher-mode.cpp)
+           as an example.
+        5. In
+           [examples/all-clusters-app/linux/main-common.cpp](https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/linux/main-common.cpp):
+            1. Add `#include "xyz-mode.h"`
+            2. In `ApplicationShutdown()`, add a call to your
+               `Clusters::XYZMode::Shutdown();`.
+        6. In
+           [examples/all-clusters-app/linux/BUILD.gn](https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/linux/BUILD.gn),
+           add the `xyz-mode.cpp` file to the `sources` section.
+        7. In
+           [src/app/common/templates/config-data.yaml](https://github.com/project-chip/connectedhomeip/blob/master/src/app/common/templates/config-data.yaml):
+            1. Add a `<XYZMode>::ModeTag` to the `EnumsNotUsedAsTypeInXML`
+               section.
+            2. Add an `XYZ Mode` entry to the
+               `CommandHandlerInterfaceOnlyClusters` section.
+        8. In
+           [src/app/util/util.cpp](https://github.com/project-chip/connectedhomeip/blob/master/src/app/util/util.cpp),
+           in the `// Cluster Init Functions...` section, add a void
+           `MatterXYZModePluginServerInitCallback() {}` definition.
+        9. In
+           [src/app/zap-templates/zcl/zcl-with-test-extensions.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/zcl-with-test-extensions.json):
+            1. Add the `xyz-mode-cluster.xml` to the `xmlFile` list
+            2. In the `attributeAccessInterfaceAttributes` entry, add your new
+               entry
+               `"XYZ Mode": [ "SupportedModes", "CurrentMode", "FeatureMap" ]` -
+               this will mean ZAP won't autogen code to handle these attributes
```

Comment:
```suggestion
               this will mean ZAP won't generate code to handle these attributes
```

And then you do not need to claim `autogen` is a word.